src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -1003,6 +1200,16 @@ private static void NarrowFourUtf16CharsToAsciiAndWriteToBuffer(ref byte outputB
                 Vector128<uint> vecNarrow = Sse2.PackUnsignedSaturate(vecWide, vecWide).AsUInt32();
                 Unsafe.WriteUnaligned<uint>(ref outputBuffer, Sse2.ConvertToUInt32(vecNarrow));
             }
+            else if (AdvSimd.Arm64.IsSupported)
+            {
+                // Narrows a vector of words [ w0 w1 w2 w3 ] to a vector of bytes
+                // [ b0 b1 b2 b3 b0 b1 b2 b3 ], then writes 4 bytes (32 bits) to the destination.
+
+                Vector128<short> vecWide = Vector128.CreateScalar(value).AsInt16();
```

Comment:
This could be simplified down to 
```c#
Vector128<short> vecWide = Vector128.CreateScalarUnsafe(value).AsInt16();
Vector64<byte> lower = AdvSimd.ExtractNarrowingSaturateUnsignedLower(vecWide);
AdvSimd.StoreSelectedScalar(ref outputBuffer, lower.AsUInt32(), 0);
```
and you don't need a call to ExtractNarrowingSaturateUnsignedUpper

Reply:
StoreSelectedScalar only takes in a byte*. I think I'd have to use Unsafe code for this?

Reply:
@pgovind It takes a pointer address, correct. But its type will depend on the type of the second arg (i.e. SIMD value)